packages/fxa-payments-server/src/lib/stripe.test.ts

```diff
@@ -2,10 +2,101 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { localeToStripeLocale } from './stripe';
+jest.mock('./account', () => ({
+  handlePasswordlessSignUp: jest.fn(),
+}));
+jest.mock('./apiClient');
+
+import * as apiClient from './apiClient';
+import { FXA_SIGNUP_ERROR, handlePasswordlessSignUp } from './account';
+import { handlePasswordlessSubscription, localeToStripeLocale } from './stripe';
+import { NEW_CUSTOMER, PLAN } from './mock-data';
+
+const stripeOverride = {
+  createPaymentMethod: jest.fn().mockResolvedValue({ paymentMethod: {} }),
+  confirmCardPayment: jest.fn().mockResolvedValue({}),
+};
+const apiClientOverrides = {
+  apiCreateCustomer: jest.fn().mockResolvedValue(NEW_CUSTOMER),
+  apiCreateSubscriptionWithPaymentMethod: jest.fn().mockResolvedValue({
+    latest_invoice: {
+      payment_intent: {
+        status: 'succeeded',
+      },
+    },
+  }),
+  apiRetryInvoice: jest.fn().mockResolvedValue({
+    payment_intent: {
+      status: 'succeeded',
+    },
+  }),
+  apiDetachFailedPaymentMethod: jest.fn().mockResolvedValue({}),
+};
 
 // handleSubscriptionPayment and handlePaymentIntent are tested as part of SubscriptionCreate
 
+// At the time of writing, there is no UI component that uses
+// handlePasswordlessSubscription; once we have that implementation, we should
+// test handlePasswordlessSubscription in the same fashion that we test
+// handleSubscriptionPayment.
```

Comment:
Maybe add a "TODO #9358" in the comment?

In general, I know this PR is part of closing the ticket above -- what all else is left? Here's what I have:
* Write similar Paypal function to `handlePasswordlessSubscription`
* Create new onSubmit handlers for Stripe and Paypal in the Checkout route that wrap these functions (this is where my newsletter function will go from PR#10031) and pass them in as the `onSubmit` prop into `PaymentForm` and `PaypalButton`.
* Replace the `/account/stub` endpoint with the final route name and test the end-to-end integration?